capstone-frontend/src/components/BookDescriptionOverlay.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Popover, OverlayTrigger } from 'react-bootstrap';
+
+const popoutDescription = (book) => {
+  let description = book.description;
+  if (description === undefined) {
+    description = "";
+  }
+  if (description.length > 500) {
+    description = description.substring(0, 500).concat('...');
+  }
+
+  return (
+    <Popover>
+      <Popover.Title as='h3'>
+        <a className='center-horizontal'
+          href={`/bookpage/${book.id}`} target='_blank'
+          rel='noopener noreferrer'>
+          {book.title}
+        </a>
+      </Popover.Title>
+      {description.length > 0 &&
```

Comment:
Seems like you could let `description` be `undefined` earlier and change this expression to take that into account (ie change to `description && ...`), rather than making it an empty string, only to not display it.

Reply:
I like this change!